### PR TITLE
Add a reverse command

### PR DESCRIPTION
**Pull request by kraetzin**
Referencing issue #442.

This implements the `reverse`, `reverse <number>-<number>`, and `reverse all` commands for editing playlists. `reverse all` is especially useful for #472 since it moves the latest played songs to the start of history.

### CODE REVIEW

File: mps_youtube/main.py [hunk 1]
```diff
@@ -2619,6 +2619,43 @@ def shuffle_fn():
...
+@commands.command(r'reverse all')
+def reverse_playlist():
+    """ Reverse order of entire loaded playlist. """
+    # Prevent crash if no last query
+    if g.last_search_query == (None, None):
+        g.content = logo()
+        g.message = "No playlist loaded"
+        return
+
+    songs_list_or_func = g.last_search_query[1]['func']
```
Comment (ids1024) [inline mps_youtube/main.py]: This crashes with `KeyError: 'func'` for a playlist search (e.g. `//playlist`).

Comment (kraetzin) [inline mps_youtube/main.py]: I hadn't considered playlist searches. I'll see if I can get those working.

**Comment by kraetzin** [comment]
I've handled the error when trying to reverse a list of playlists. I also handled an error that occurs when trying to reverse the list of user uploads.

**Comment by ids1024** [comment]
> I also handled an error that occurs when trying to reverse the list of user uploads.

I made this actually work in commit f439c1f6c80c3b98549f92235e0f7b2b0132f59e.

If `paginatesongs`, `IterSlicer` etc. seem somewhat convoluted, they sort of are; when  #423 is done the API for these things should be a lot cleaner.

**Pull request merged**